app/backend-dotnet/Program.cs

```diff
@@ -0,0 +1,72 @@
+using Azure;
+using Azure.AI.OpenAI;
+using Azure.Core;
+using Azure.Identity;
+using Azure.Search.Documents;
+using Azure.Storage.Blobs;
+using IMJohn.Services;
+using Microsoft.Extensions.FileProviders;
+using Microsoft.SemanticKernel;
+using Microsoft.SemanticKernel.Configuration;
+
+var builder = WebApplication.CreateBuilder(args);
+
+// Add services to the container.
+
+builder.Services.AddControllers();
+// Learn more about configuring Swagger/OpenAPI at https://aka.ms/aspnetcore/swashbuckle
+builder.Services.AddEndpointsApiExplorer();
+builder.Services.AddSwaggerGen();
+
+var azureCredential = new DefaultAzureCredential();
+var AZURE_STORAGE_ACCOUNT = builder.Configuration.GetValue<string>("AZURE_STORAGE_ACCOUNT");
+
+// Add blob service client
+var blobServiceClient = new BlobServiceClient(new Uri($"https://{AZURE_STORAGE_ACCOUNT}.blob.core.windows.net"), azureCredential);
+builder.Services.Add(ServiceDescriptor.Singleton(blobServiceClient));
+
+// Add blob container client
+var AZURE_STORAGE_CONTAINER = builder.Configuration.GetValue<string>("AZURE_STORAGE_CONTAINER");
+var blobContainerClient = blobServiceClient.GetBlobContainerClient(AZURE_STORAGE_CONTAINER);
+builder.Services.Add(ServiceDescriptor.Singleton(blobContainerClient));
```

Comment:
```suggestion
builder.Services.AddSingleton<BlobContainerClient>(provider =>
{
    var config = provider.GetRequiredServices<IConfiguration>();
    var storageContainer = config.GetValue<string>("AZURE_STORAGE_CONTAINER");
    var serviceClient = provider.GetRequiredServices<BlobServiceClient>();
    
    return serviceClient.GetBlobContainerClient(storageContainer );
});
```

Again, I'd likely rewrite this to avoid capturing over a closure and encapsulating the registration logic. This could more easily be refactored into an extension method later too. I have the same recommendations for several related refactorings in this file.